packages/grafana-ui/src/components/MatchersUI/FieldNameMatcherEditor.tsx

```diff
@@ -15,7 +15,7 @@ export const FieldNameMatcherEditor = memo<MatcherUIProps<string>>(props => {
       }
       return props.onChange(selection.value);
     },
-    [names, props.onChange]
+    [names, props]
```

Comment:
nit: Best practices is to destruct nested props and use them in the dependencies.

Reply:
done